src/Sarif.Multitool.Library/Rules/SARIF1010.RuleIdMustBeConsistent.cs

```diff
@@ -26,6 +26,12 @@ public class RuleIdMustBeConsistent : SarifValidationSkimmerBase
         };
         public override FailureLevel DefaultLevel => FailureLevel.Error;
 
+        private List<string> ruleIds;
```

Comment:
I can see some issues:
1. Imagine that the sarif you are validating has two runs. In each run you have rules and results. For the first run, we have rule01 and result01 (fine). for the second run, we can have rule02' and result01', but from a different tool, so by coincidence you are saying it's right but it isn't because the result01' should be looking for rule01'.
2. If you have multiple equal rules registered, your list will be big. Probably you should change it to a Dict<toolname,hashset>, with that, you would prevent the first issue and it would minimize the search/duplication